cli/lsp/completions.rs

```diff
@@ -452,6 +471,134 @@ fn get_relative_specifiers(
     .collect()
 }
 
+/// Get completions for `npm:` specifiers.
+async fn get_npm_completions(
+  specifier: &str,
+  range: &lsp::Range,
+  npm_registry_url: &Url,
+  file_fetcher: &FileFetcher,
```

Comment:
Instead of providing a file fetcher and registry url, maybe we should create and provide an `NpmSearchApi` struct here instead that would have the file fetcher and `npm_registry_url` as being internal to it. Then that would allow us to easily create an `NpmSearchApi` trait (with `RealNpmSearchApi` struct) that we could then easily implement for testing purposes.

Reply:
I merged it into `CliNpmRegistryApi` after all.